pkg/cri/server/sandbox_run_linux.go

```diff
@@ -101,12 +101,13 @@ func (c *criService) sandboxContainerSpec(id string, config *runtime.PodSandboxC
 	if nsOptions.GetIpc() == runtime.NamespaceMode_NODE {
 		sandboxDevShm = devShm
 	}
+	specOpts = append(specOpts, oci.WithoutMounts("/dev/shm"))
 	specOpts = append(specOpts, oci.WithMounts([]runtimespec.Mount{
 		{
 			Source:      sandboxDevShm,
 			Destination: devShm,
 			Type:        "bind",
-			Options:     []string{"rbind", "ro"},
+			Options:     []string{"rbind", "ro", "nosuid", "nodev", "noexec"},
```

Comment:
I think it would be better to change it for all. Otherwise we will get an `/dev/shm` with:
```
shm on /dev/shm type tmpfs (ro,relatime,size=65536k)
```
which doesn't confirm to the runC spec(https://github.com/opencontainers/runc/blob/main/libcontainer/SPEC.md):
```
/dev/shm | tmpfs | MS_NOEXEC,MS_NOSUID,MS_NODEV | mode=1777,size=65536k
```

Reply:
@jiangliu Well, it is a backwards incompatible change, so IMHO we should put some thought into doing that change. At least, IMHO, it deserves to be highlighted in the PR description for open debate, and clear reasons to why it is worth doing it.

> which doesn't confirm to the runC spec(https://github.com/opencontainers/runc/blob/main/libcontainer/SPEC.md):

Do we care about that? I mean, AFAIK we should care about the runtime-spec. That seems to be some documentation for the libcontainer and its defaults. It is good to know, but nothing else IMHO.

Reply:
BTW, we found this issue when analyzing the generated OCI runtime config file and then found the root cause.

Reply:
> https://github.com/kinvolk/containerd/blob/54c1f3529d64b58e187ab0341b3b8f669be30a0d/pkg/cri/server/sandbox_run_linux.go#L122-L134

For user ns enabled case, we could also enable the "ro" flag.
```
	sanboxDevShmOpt := []string{"rbind", "ro"}
	if userNS {
		// When userns is enabled, the mount fails if it doesn't include these options,
		// except for rprivate. rpivate is not required, but nice to have.
		sanboxDevShmOpt = append(sanboxDevShmOpt, "rprivate", "noexec", "nosuid", "nodev")
	}
```

Reply:
Ok, so I guess due to the lack of comment on that part, that the runc/libcontainer thing is ok, right?

I'm in a rush for kubecon and to make sure the KEP makes it into 1.25. I don't have time to review/test this now. However, _if this only affects the sandbox mount_, it is probably not a big issue. I don't see any good reason to do it one way or the other. But it is still worth noting this change in the PR description and release notes IMHO.